stdlib/multiprocessing/forkserver.pyi

```diff
@@ -0,0 +1,44 @@
+from _typeshed import FileDescriptorLike, StrOrBytesPath
+from collections.abc import Iterable, Mapping, Sequence
+from signal import _HANDLER, _SIGNUM
+from struct import Struct
+from threading import Lock
+from typing import Any
+
+__all__ = ["ensure_running", "get_inherited_fds", "connect_to_new_process", "set_forkserver_preload"]
+
+MAXFDS_TO_SEND: int = ...
+SIGNED_STRUCT: Struct = ...
+
+class ForkServer:
+    _forkserver_address: StrOrBytesPath | None
+    _forkserver_alive_fd: int | None
+    _forkserver_pid: int | None
+    _inherited_fds: Sequence[int] | None
+    _lock: Lock
+    _preload_modules: list[str]
+
+    def __init__(self) -> None: ...
+    def _stop(self) -> None: ...
+    def _stop_unlocked(self) -> None: ...
+    def set_forkserver_preload(self, modules_names: list[str]) -> None: ...
+    def get_inherited_fds(self) -> Sequence[int] | None: ...
+    def connect_to_new_process(self, fds: Sequence[int]) -> tuple[int, int]: ...
+    def ensure_running(self) -> None: ...
+
+def main(
+    listener_fd: int | None,
+    alive_r: FileDescriptorLike,
+    preload: Sequence[str],
+    main_path: str | None = ...,
+    sys_path: object | None = ...,
+) -> None: ...
+def _serve_one(child_r: int, fds: Iterable[int], unused_fds: Iterable[int], handlers: Mapping[_SIGNUM, _HANDLER]) -> Any: ...
+def read_signed(fd: int) -> Any: ...
+def write_signed(fd: int, n: Any) -> None: ...
```

Comment:
`SIGNED_STRUCT` has `'q'` as the `format_value` passed to its `__init__` (https://github.com/python/cpython/blob/2833f3798dc3647e850b303a4d0fa00609a0ae9b/Lib/multiprocessing/forkserver.py#L26), and that means that only `int`s can be passed to its `.pack` method (docs here: https://docs.python.org/3/library/struct.html#format-characters)

```suggestion
def write_signed(fd: int, n: int) -> None: ...
```

Reply:
Ah, sorry about that. Thank you for explaining how you got to that conclusion!

Reply:
> Ah, sorry about that. Thank you for explaining how you got to that conclusion!

No need to be sorry! Typing is hard :)

If the reason for something one of us is saying is every unclear, please don't be afraid to ask for clarification btw!